From patchwork Tue Oct  9 16:39:26 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Paolo Bonzini <pbonzini@redhat.com>
X-Patchwork-Id: 10632955
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id F093C112B
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  9 Oct 2018 16:39:38 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id C5173291B2
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  9 Oct 2018 16:39:38 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id B852E291BE; Tue,  9 Oct 2018 16:39:38 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-6.1 required=2.0 tests=BAYES_00,DKIM_INVALID,
	DKIM_SIGNED,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI,URIBL_SBL autolearn=ham
	version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 2E63B291B2
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  9 Oct 2018 16:39:32 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726840AbeJIX5T (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 9 Oct 2018 19:57:19 -0400
Received: from mail-wm1-f68.google.com ([209.85.128.68]:53389 "EHLO
        mail-wm1-f68.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1726481AbeJIX5T (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 9 Oct 2018 19:57:19 -0400
Received: by mail-wm1-f68.google.com with SMTP id y11-v6so2551316wma.3;
        Tue, 09 Oct 2018 09:39:30 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20161025;
        h=sender:from:to:subject:date:message-id;
        bh=r1bccujBhggz+nVVtPgQWsjosGEPtlkDO0q0NXskKMU=;
        b=C1GO9pUAZtqOj1SW67zFLWYOLJZLdecV4zp8cHqr+LCX6Fk8gPJPTthYX91di1kR76
         pVyrVJPj+TucAzKnosrNO/UVd2I3GsXDiUxKUExWa661/5Pm5i9PYlMBDb5GoPIY4mLY
         K8p7uUzsODh67rz0bd1MfyXUSGyCLBRHO4+QbXbIjJh9Mb+O8w2rdQmy59OCQzen4wVS
         L28MufRPIM0u+s4qhavJcr7NkIrYTCdKyOpXjd1gabJ9aSouZuUFH9Soz4aWhd5E4kfU
         z7xUYbgctfibOc3qRDo6DvHZb8kzh9TLPNurlim84S2fCqT68QBMafAPtU/uzHwfZfD3
         GZdQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:sender:from:to:subject:date:message-id;
        bh=r1bccujBhggz+nVVtPgQWsjosGEPtlkDO0q0NXskKMU=;
        b=bfCMWlmnNv13u/i3z9lwjPfBwcD1z/3FbsO4sW7/W00nMCDpFliY0BjuzV7FFtnS9H
         IYWexC7Z4gneaalmv48wVMDbFrV/B7NXfJlNUVNJZeI8Awn75KMXnyncfWvguZ21YP/Y
         z/3ZLLIK1PUsAwlQqNtic58QiKMcz72/XokVvTUXaGS0AOkdqWF5r9GQ1IAWQSpVvBzp
         5+0g67wFGBBoIYJrV0lhJJehkGYmfu4fQVQImMzWNeckGpY8TK3lCx2LGxGNhrKZ/7yd
         iz16giCrVSU4gq1RsXHGqz0/xRnv66pyrODYwOSSjyt8Lblzh+NoHoaUFsrMUVTY4stX
         tWDQ==
X-Gm-Message-State: ABuFfogRXaw1qOnJAyExGZSWATV8ud6q3DDDwwIJG4++VhcTNEy93+gR
        aaNvYoBydhl/BxY/oyhLZXrmklYc
X-Google-Smtp-Source: 
 ACcGV63BSxOHpz7dZU8mFltGcvOKYMh3vsr7OruMwIjdYJS4iZbgedLC0T3ZPKr/oho/r5lFyk1rZg==
X-Received: by 2002:a1c:8e8c:: with SMTP id
 q134-v6mr2589281wmd.112.1539103169150;
        Tue, 09 Oct 2018 09:39:29 -0700 (PDT)
Received: from 640k.lan (94-36-184-18.adsl-ull.clienti.tiscali.it.
 [94.36.184.18])
        by smtp.gmail.com with ESMTPSA id
 n11-v6sm25165866wra.26.2018.10.09.09.39.27
        (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
        Tue, 09 Oct 2018 09:39:28 -0700 (PDT)
From: Paolo Bonzini <pbonzini@redhat.com>
To: linux-kernel@vger.kernel.org, kvm@vger.kernel.org
Subject: [PATCH] KVM: x86: support CONFIG_KVM_AMD=y with
 CONFIG_CRYPTO_DEV_CCP_DD=m
Date: Tue,  9 Oct 2018 18:39:26 +0200
Message-Id: <1539103166-16317-1-git-send-email-pbonzini@redhat.com>
X-Mailer: git-send-email 1.8.3.1
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

SEV requires access to the AMD cryptographic device APIs, and this
does not work when KVM is builtin and the crypto driver is a module.
Actually the Kconfig conditions for CONFIG_KVM_AMD_SEV try to disable
SEV in that case, but it does not work because the actual crypto
calls are not culled, only sev_hardware_setup() is.

This patch adds two CONFIG_KVM_AMD_SEV checks that gate all the remaining
SEV code; it fixes this particular configuration, and drops 5 KiB of
code when CONFIG_KVM_AMD_SEV=n.

Reported-by: Guenter Roeck <linux@roeck-us.net>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
---
 arch/x86/kvm/svm.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/arch/x86/kvm/svm.c b/arch/x86/kvm/svm.c
index d96092b35936..61ccfb13899e 100644
--- a/arch/x86/kvm/svm.c
+++ b/arch/x86/kvm/svm.c
@@ -436,14 +436,18 @@ static inline struct kvm_svm *to_kvm_svm(struct kvm *kvm)
 
 static inline bool svm_sev_enabled(void)
 {
-	return max_sev_asid;
+	return IS_ENABLED(CONFIG_KVM_AMD_SEV) ? max_sev_asid : 0;
 }
 
 static inline bool sev_guest(struct kvm *kvm)
 {
+#ifdef CONFIG_KVM_AMD_SEV
 	struct kvm_sev_info *sev = &to_kvm_svm(kvm)->sev_info;
 
 	return sev->active;
+#else
+	return false;
+#endif
 }
 
 static inline int sev_get_asid(struct kvm *kvm)
